common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -379,9 +399,10 @@ void deleteExecutorDirs(AppShuffleInfo appShuffleInfo) {
   @Override
   public StreamCallbackWithID receiveBlockDataAsStream(PushBlockStream msg) {
     AppShuffleInfo appShuffleInfo = validateAndGetAppShuffleInfo(msg.appId);
-    final String streamId = String.format("%s_%d_%d_%d_%d",
-      OneForOneBlockPusher.SHUFFLE_PUSH_BLOCK_PREFIX, msg.shuffleId, msg.shuffleMergeId,
-      msg.mapIndex, msg.reduceId);
+    // Use string concatenation here to avoid the overhead with String.format on every
```

Comment:
What's the exact overhead? It becomes slower?

Reply:
This is dependent on the number of blocks pushed, since this is incurred with every block pushed.
The server has no problem to keep up in our benchmark which pushed 40M blocks, but from flamegraph we could see a noticeable CPU time spent in this String.format (~5%).
For push-based shuffle, it would always be good to reduce the server side load.

Reply:
+1 to avoid `String.format` for simple patterns (unless it is being lazily evaluated on demand).